Does not work for us anymore (SpaceBot, UTM-30lx), got

```
[20151125-12:13:54:271] [ERROR] - ""::unknown error code "\n0"
 (/home/sbot/dev/spacebot_release/drivers/hokuyo/hokuyo.cc:390 - bool URG::parseErrorCode(const char*, const URG::StatusCode*))
cannot open device: UNKNOWN
errno is: Operation not supported
```

and I wonder how it should work for any scanner. urg.open() calls fullSerialReset() and within this method the for-loop returns an error or false and if we pass the loop successfully we return false..? 

``` cpp
bool URG::fullSerialReset() {
    LOG_INFO_S << "Resetting scanner..." << flush;
    size_t baudrates[]={19200, 57600, 115200};
    const int baudrates_count = 3;

    int i;
    for (i=0; i < baudrates_count; i++){
        if (!setSerialBaudrate(baudrates[i]))
            return error(URG::BAD_HOST_RATE);;

        if (!initCommunication() )
            return false;
    }

    if (i == baudrates_count)
        return false;

    if (!setSerialBaudrate(19200))
        return error(URG::BAD_HOST_RATE);;

    // Set baud rate to default
    LOG_INFO_S << " done" << endl;
    baudrate = 19200;
    return true;
}
```

--- a/hokuyo.cc
+++ b/hokuyo.cc
@@ -494,9 +494,6 @@ bool URG::fullSerialReset() {
             return false;
     }
 
-    if (i == baudrates_count)
-        return false;
-
     if (!setSerialBaudrate(19200))
         return error(URG::BAD_HOST_RATE);;
 

